lib/AIECreateLogicalLocks.cpp

```diff
@@ -0,0 +1,72 @@
+//===- AIEAssignBuffers.cpp -------------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2019 Xilinx Inc.
+//
+//===----------------------------------------------------------------------===//
+
+// This pass aims to assign lockIDs to AIE.lock operations. The lockID is
+// numbered from the most recent AIE.lock within the same tile. If the lockID
+// exceeds 15 then the pass generates an error and terminates. AIE.lock
+// operations for different tiles are numbered independently. If there exists an
+// existing lockID this pass overwrites the existing lockID generating a
+// warning.
+
+#include "aie/AIEDialect.h"
+#include "mlir/IR/Attributes.h"
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#define DEBUG_TYPE "aie-logical-locks"
+
+using namespace mlir;
+using namespace xilinx;
+using namespace xilinx::AIE;
+
+struct AIECreateLogicalLocksPass
+    : public AIECreateLogicalLocksBase<AIECreateLogicalLocksPass> {
+  void getDependentDialects(::mlir::DialectRegistry &registry) const override {
+    registry.insert<func::FuncDialect>();
+    registry.insert<xilinx::AIE::AIEDialect>();
+  }
+  void runOnOperation() override {
+
+    ModuleOp m = getOperation();
+    OpBuilder rewriter = OpBuilder::atBlockEnd(m.getBody());
+
+    // loop through locks
+    // store lockID count in map with operation as key
+    std::map<Operation *, int> unique_tiles;
```

Comment:
maybe "tileToLastID"?  This gives a little bit more information about what the Operation * is pointing to.